packages/truffle-core/lib/test.js

```diff
@@ -144,36 +145,33 @@ const Test = {
     return web3.eth.getAccounts();
   },
 
-  compileContractsWithTestFilesIfNeeded: function(
+  compileContractsWithTestFilesIfNeeded: async function(
     solidityTestFiles,
     config,
     testResolver
   ) {
-    return new Promise(function(accept, reject) {
-      Profiler.updated(config.with({ resolver: testResolver }), function(
-        err,
-        updated
-      ) {
-        if (err) return reject(err);
-
-        updated = updated || [];
-
-        const compileConfig = config.with({
-          all: config.compileAll === true,
-          files: updated.concat(solidityTestFiles),
-          resolver: testResolver,
-          quiet: config.quiet,
-          quietWrite: true
-        });
-        // Compile project contracts and test contracts
-        Contracts.compile(compileConfig)
-          .then(result => {
-            const paths = result.outputs.solc;
-            accept(paths);
-          })
-          .catch(reject);
-      });
+    const updated =
+      (await promisify(Profiler.updated)(
```

Comment:
No need for this `promisify`. `Profiler.updated` now has a dual Promise/callback interface. Check it [out](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-compile/profiler/index.js#L19).

Reply:
Hot diggity!